### DISCUSSION TITLE
Silent walking

### DISCUSSION THREAD

**Discussion by AdamTadeusz** [NeotokyoRebuild/neo#623] [Ideas]
With how strong sound is in the new engine, making it very easy to locate where people are coming from, I think making the walk feature useful by stopping player footstep sounds when held should be considered

**Comment by kassibuss** [comment]
Does it not already do that?

**Reply by AdamTadeusz** [reply]
pressing the walk button slows you down, but whether you make footsteps or not depends entirely on your velocity, and holding alt does not guarantee your velocity will fall below the no footsteps threshold. Another thing to keep in mind is whether its possible to bhop while walking, we probably don't want players to be able to bhop at excessive speeds without making any noise at all. If that is the case, then perhaps simply raising the no footsteps threshold so holding alt guarantees you don't make footsteps regardless of class unless you try to bunnyhop

**Reply by AdamTadeusz** [reply]
keep in mind things like changing direction increases your velocity too. You can for example walk silently as a support when holding alt and moving in one direction only, but as soon as you change your direction of movement slightly your character will make footsteps because you speed up slightly

**Reply by AdamTadeusz** [reply]
If we were simply to increase the footsteps threshold then accounting for the slightly faster speeds when turning would be good too. Perhaps different thresholds depending on class would be required

**Comment by Rainyan** [comment]
Slightly adjacent to this topic, I'd be in favor of a high dynamic cutoff for distance based cues, i.e. make sounds roughly fit in 3 categories:

* obviously audible
* slightly quieter, but still audible if you pay attention/aren't distracted by other simultaneously playing sounds
* completely silent (no signal whatsoever)

This would have the benefit of removing the kind of prisoner's dilemma, whereby you can gain a competitive edge by playing at unhealthy volumes, where simply cranking your volume levels really high would net you extra info at the expense of hearing damage when actually LOUD sounds play.

**Reply by AdamTadeusz** [reply]
I absolutely agree, adding some sound cutoff based on distance would be a good idea, need to investigate to see if the sound emitter system supports something like that. If I remember correctly you can pass a filter to decide who hears a sound so even if its not natively supported, might just be a case of making a custom filter which filters out players based on distance

**Reply by AdamTadeusz** [reply]
https://github.com/user-attachments/assets/e958811a-87b4-480a-b587-7b519c2f9705

Here's what a footstep sound cutoff of 500 hu would look and sound like

**Comment by blaberry** [comment]
I think going for rains intent and adjusting velocity thresholds upwards is a good idea, it keeps with the spirit of the original while compensating for the increase in utility audio has in 2013.
